documentation/specs/static-graph.md

```diff
@@ -246,4 +246,31 @@ In fact, this same mechanism can be used for non-distributed builds as well, for
 **WIP**
 
 ## I/O Tracking
-**WIP**
+To help facilitate caching of build outputs by a higher-order build engine, MSBuild needs to track all I/O that happens as part of a build.
+
+**OPEN ISSUE:** This isn't actually true in most scenarios. Today the MS internal build engine can wrap any arbitrary process to track the I/O that happens as part of its execution as well as its children. That's sufficient for all scenarios except compiler servers or an MSBuild server (see below).
```

Comment:
Maybe worth summarizing the `problem`: if BuildXL wants to adapt other build engines, then there's the problem of tracking requirements mismatch between engines, and long lived build services